src/main/java/com/rackspace/feeds/feedscatalog/ResolveHostFilter.java

```diff
@@ -0,0 +1,141 @@
+
+    static Logger LOG = LoggerFactory.getLogger(ResolveHostFilter.class);
+    static final String resolveHostXslt = "feedscatalog-resolve-host.xsl";
+
+    private TransformerUtils transformer;
+
+    public void init(FilterConfig filterConfig) throws ServletException {
+        LOG.debug( "initializing ResolveHostFilter" );
+
+        try {
+            //load the resolve host xslt from resource
+            URL xsltResource = Thread.currentThread().getContextClassLoader().getResource(resolveHostXslt);
+            if (xsltResource != null) {
+                transformer = TransformerUtils.getInstanceForXsltAsFile(xsltResource.getPath());
+            }
+            else {
+                throw new UnavailableException("resource is null - " + resolveHostXslt);
+            }
+        } catch ( Exception e ) {
+            LOG.error( "Error loading xslt: " + e.getMessage());
+            throw new ServletException( e );
+        }
+    }
+
+    public void doFilter (ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+
+        HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+        HttpServletResponse httpServletResponse = (HttpServletResponse) response;
+
+        // setup wrapper response with output stream to collect response content
+        StringResponseWrapper wrappedResponse = new StringResponseWrapper(httpServletResponse);
+
+        // apply filter further down the chain on wrapped response
+        chain.doFilter(httpServletRequest, wrappedResponse);
+
+        // obtain response content
+        String originalResponseContent = wrappedResponse.getResponseString();
+
+        // transform the response with resolve-host if not empty
+        if (StringUtils.isNotEmpty(originalResponseContent)) {
+
+            // create input params to xslt with headers from request
+            HashMap<String, Object> params = new HashMap<String, Object>();
+            putTenantIdParams(httpServletRequest, params);
+            putExternalLocParams(httpServletRequest, params);
+
+            try {
+                // create outputStream to store result of transformation
+                OutputStream outputStream = new ByteArrayOutputStream();
+
+                // transform response content with resolve-host xslt
+                transformer.doTransform(params,
+                    new StreamSource(new StringReader(originalResponseContent)),
+                    new StreamResult(outputStream));
+
+                // set transformed content to response
+                String newResponseContent = outputStream.toString();
+                httpServletResponse.setContentLength(newResponseContent.length());
+                httpServletResponse.getWriter().write(newResponseContent);
+            }
+            catch (Exception e) {
+                LOG.error("Error transforming xml:" + e.getMessage());
+                throw new ServletException(e);
```

Comment:
I am not a big fan of wrapping exceptions if we are not adding any additional information. Why don't you just not catch it and let the Servlet engine catch it?